scripts/doc_checker.py

```diff
@@ -533,6 +533,13 @@ def load_obj(name, old_load_obj=Docstring._load_obj):
     # enable docs testing on windows
     sys.getdlopenflags = Mock()
     sys.setdlopenflags = Mock()
+    xgboost_mock = Mock()
+
+    class Booster:
+        ...
+
+    xgboost_mock.Booster = Booster
```

Comment:
Otherwise, the documentation check will break, since the object inheriting (`Booster(xgb.Booster)`) from the mock itself becomes a mock. The functionality of the inspect module does not work with mock objects:
```bash
Miniconda3\envs\test-modin\lib\inspect.py", line 676, in getfile
    raise TypeError('module, class, method, function, traceback, frame, or '
TypeError: module, class, method, function, traceback, frame, or code object was expected, got Mock
```